tools/fetch-docs-links.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * @fileoverview Script to fetch link data.
+ *
+ * To fetch info about all files:
+ *
+ *      node tools/fetch-docs-links.js
+ *
+ * To fetch info for just selected files (for use with lint-staged):
+ *
+ *      node tools/fetch-docs-links.js docs/src/user-guide/index.md
+ *
+ * @author Nicholas C. Zakas
+ */
+
+"use strict";
+
+//-----------------------------------------------------------------------------
+// Requirements
+//-----------------------------------------------------------------------------
+
+const matter = require("gray-matter");
+const metascraper = require("metascraper")([
+    require("metascraper-image")(),
+    require("metascraper-logo")(),
+    require("metascraper-logo-favicon")(),
+    require("metascraper-title")(),
+    require("metascraper-description")()
+]);
+const got = require("got");
+const path = require("path");
+const fs = require("fs/promises");
+const glob = require("fast-glob");
+
+//-----------------------------------------------------------------------------
+// Data
+//-----------------------------------------------------------------------------
+
+const BASE_DIR = path.resolve(__dirname, "../");
+const SRC_DIR = path.resolve(BASE_DIR, "docs/src");
+const DATA_DIR = path.resolve(SRC_DIR, "_data");
+const DATA_FILE_PATH = path.resolve(DATA_DIR, "further_reading_links.json");
+
+// determine which files to check
+let filenames = process.argv.slice(2);
+
+if (filenames.length === 0) {
+    filenames = glob.sync("docs/src/rules/*.md", { cwd: BASE_DIR });
+}
+
+filenames = filenames.map(filename => path.resolve(BASE_DIR, filename));
+
+//-----------------------------------------------------------------------------
+// Helpers
+//-----------------------------------------------------------------------------
+
+/**
+ * Fetches metadata information for a given URL.
+ * @param {string} url The URL to fetch data for.
+ * @returns {Promise<object>} An object with metadata info.
+ */
+async function fetchLinkMeta(url) {
+    const { body: html } = await got(url);
+    const metadata = await metascraper({ html, url });
```

Comment:
I think we should pass to `metascraper()` the url that `got` returns (as in the previous implementation), as recommended in the [metascraper usage example](https://metascraper.js.org/#/?id=usage), in order to get urls for resources (logo) used by the final document after `got` resolves all redirects.

```suggestion
    const { body: html, url: returnedURL } = await got(url);
    const metadata = await metascraper({ html, url: returnedURL });
```

It might make sense to use the same url  when calculating the value of the `domain` variable below, too.